### ISSUE TITLE
Fernet key not the same in the different airflow containers

### ISSUE THREAD

**Issue by tylermichael** [astronomer/astro-cli#123]
The fernet key on the webserver and scheduler are different so when I try to use an encrypted connection, I get this error:

```
[2018-09-20 15:24:12,219] {models.py:1736} ERROR - 
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/cryptography/fernet.py", line 108, in _decrypt_data
    h.verify(data[-32:])
  File "/usr/lib/python3.6/site-packages/cryptography/hazmat/primitives/hmac.py", line 69, in verify
    ctx.verify(signature)
  File "/usr/lib/python3.6/site-packages/cryptography/hazmat/backends/openssl/hmac.py", line 73, in verify
    raise InvalidSignature("Signature did not match digest.")
cryptography.exceptions.InvalidSignature: Signature did not match digest.
```

An easy fix might be to add a generated fernet key to the `ComposeConfig` here and in the yml template as the env var `AIRFLOW__CORE__FERNET_KEY`: [here](https://github.com/astronomerio/astro-cli/blob/9cfc141a853681a6818073cb825987b2189cef74/airflow/docker.go#L84)

When I get in the containers and look, they do have values in their airflow configs for `fernet_key` but I'm not sure where those are coming from.

I totally could be missing something obvious here so any help would be appreciated!

**Comment by andscoop** [comment]
@tylermichael Which images are you using? Currently when I spin up a clean project and run 

`docker exec testproj_scheduler_1 cat airflow.cfg | grep -i fernet`
`docker exec testproj_webserver_1 cat airflow.cfg | grep -i fernet`

to view the scheduler and webserver fernet keys respectively I am showing matching fernet keys across airflow images.

----

If you want to provide a custom fernet key, you can specify the AIRFLOW__CORE__FERNET_KEY ENV in the Dockerfile. This will have the effect of overriding the config across both containers.

**Comment by tylermichael** [comment]
Heh 😄 I actually ended up forking the Astronomer repo where the airflow images are being built and built the image for Airflow 1.10 because I needed some feature in that version. I realize that it may not be in a ready state as version 1.10 was commented out in the build scripts.

I had to do that because my requirements file has airflow and some other dependencies defined in it so that I get intellisense in my editor, but I had to do a little bit of hack to get that working on my end.

I'm thinking I had to do my hack because version 1.10 hasn't been officially built and released by you guys.

Do you know how the keys ended up being the same? I couldn't find where that was happening to see if my changes effected that at all.

---
That's a great idea overriding it in the Dockerfile. I'll try that out.

Thanks!

**Comment by tylermichael** [comment]
Setting that in the Dockerfile did fix my issue.

**Comment by andscoop** [comment]
@tylermichael glad to hear it! I just wanted to follow up, do you know if you were running into use-case #5, listed in [this document](https://github.com/astronomerio/incubator-airflow/blob/94e7ded882305eec1db36cafaa4f70d61a2f12aa/docs/howto/secure-connections.rst#L32)?

----

I am still digging in on how the FERNET_KEYs end up being the same. If you are able to share your project Dockerfiles with any sensitive information removed, it would help the troubleshooting.

[Here](https://github.com/astronomerio/incubator-airflow/blob/94e7ded882305eec1db36cafaa4f70d61a2f12aa/airflow/configuration.py#L56) is where the keys are being generated

**Comment by tylermichael** [comment]
I did not run into that use case. This was happening on a new project.

The only Dockerfile change I made was here: https://github.com/astronomerio/astronomer/blob/3ceb6d234631d616ae6507ba6b5ee6b532d4490a/docker/airflow/1.10.0/onbuild/Dockerfile#L16

I made the onbuild image use the built 1.10 image as the base.
When I went to build my image for 1.10 it ended up using the old 1.9 image on my machine because it happened to have the same name as what the 1.10 image is looking for, `astronomerinc/ap-airflow`.

So when my `requirements.txt` file was being installed, that GPL env var wasn't present for airflow so the pip install would fail, because it was using the old 1.9 image.

This happened because the build script doesn't explicitly tag the "new" image as `latest` here: https://github.com/astronomerio/astronomer/blob/3ceb6d234631d616ae6507ba6b5ee6b532d4490a/bin/build-airflow#L15

Hopefully that makes sense....
